mobly/base_test.py

```diff
@@ -1014,21 +1017,53 @@ def _get_test_methods(self, test_names):
         This can only be caused by user input.
     """
     test_methods = []
+    # Call once and reuse the list through this function for efficiency.
+    current_valid_names = self.get_existing_test_names()
+    # Process the test name selector one by one.
     for test_name in test_names:
-      if not test_name.startswith('test_'):
-        raise Error(
-            'Test method name %s does not follow naming '
-            'convention test_*, abort.' % test_name
+      if test_name.startswith(TEST_SELECTOR_REGEX_PREFIX):
+        # process the selector as a regex.
+        regex_matching_methods = self._get_regex_matching_test_methods(
+            test_name[len(TEST_SELECTOR_REGEX_PREFIX) :], current_valid_names
         )
+        test_methods += regex_matching_methods
+        continue
+      # process the selector as a regular test name string.
+      self._assert_valid_test_name(test_name)
+      if test_name not in current_valid_names:
+        raise Error('%s does not have test method %s.' % (self.TAG, test_name))
       if hasattr(self, test_name):
         test_method = getattr(self, test_name)
       elif test_name in self._generated_test_table:
         test_method = self._generated_test_table[test_name]
-      else:
-        raise Error('%s does not have test method %s.' % (self.TAG, test_name))
       test_methods.append((test_name, test_method))
     return test_methods
 
+  def _get_regex_matching_test_methods(
+      self, test_name_regex, current_valid_names
+  ):
+    matching_name_tuples = []
```

Comment:
The constant name seems not accurately reflect its type. Should we rename it to `matching_name_lists`?

Reply:
sorry, can you clarify?
this is a list of tuples, no?

Reply:
Ah, got it. At first glance, it looks like a tuple type from its name.